sopel/plugins/rules.py

```diff
@@ -1196,15 +1241,15 @@ def get_rule_regex(self):
         * the rule's name (escaped for regex),
         * all of its aliases (escaped for regex),
 
-        and everything is then given to
-        :func:`sopel.tools.get_nickname_command_regexp`, which creates a
-        compiled regex to return.
+        to creates a compiled regex to return.
         """
         name = [self.escape_name(self._name)]
         aliases = [self.escape_name(alias) for alias in self._aliases]
         pattern = r'|'.join(name + aliases)
-        return tools.get_nickname_command_regexp(
-            self._nick, pattern, self._nick_aliases)
+        return tools.compile_rule(
+            self._nick,
+            self.PATTERN_TEMPLATE.format(command=pattern),
+            self._nick_aliases)
```

Comment:
I must have forgotten to mention it in that issue you were working from, but the new rule system is a perfect excuse to deprecate `tools.compile_rule()` too, and move its behavior into the `Rule` type.

Reply:
Hm. It requires a bit more work. I shall do that in another PR :tm:

Reply:
I'll probably make an issue for this, because with no plans to release any 7.2 branch, our last chance to deprecate anything else we want to remove in 8.0 (like `tools.compile_rule`) is 7.1.